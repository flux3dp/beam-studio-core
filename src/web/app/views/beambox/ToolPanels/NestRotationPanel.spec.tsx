/* eslint-disable import/first */
import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

jest.mock('helpers/i18n', () => ({
  lang: {
    beambox: {
      tool_panels: {
        _nest: {
          rotations: 'Possible Rotation',
        },
      },
    },
  },
}));

import NestRotationPanel from './NestRotationPanel';

test('should render correctly', () => {
  const onValueChange = jest.fn();
  const wrapper = shallow(<NestRotationPanel
    rotations={1}
    onValueChange={onValueChange}
  />);
  expect(toJson(wrapper)).toMatchSnapshot();

  wrapper.find('UnitInput').props().getValue(2);
  expect(onValueChange).toHaveBeenCalledTimes(1);
  expect(onValueChange).toHaveBeenNthCalledWith(1, 2);
  expect(toJson(wrapper)).toMatchSnapshot();
});
