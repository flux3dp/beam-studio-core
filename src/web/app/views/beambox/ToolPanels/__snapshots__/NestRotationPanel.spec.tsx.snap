// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="tool-panel"
>
  <label
    className="controls accordion"
  >
    <input
      className="accordion-switcher"
      defaultChecked={true}
      type="checkbox"
    />
    <p
      className="caption"
    >
      Possible Rotation
      <span
        className="value"
      >
        1
      </span>
    </p>
    <label
      className="accordion-body"
    >
      <div>
        <div
          className="control nest-rotations"
        >
          <UnitInput
            abbr={false}
            className={Object {}}
            decimal={0}
            defaultValue={1}
            disabled={false}
            displayMultiValue={false}
            forceUsePropsUnit={false}
            getValue={[Function]}
            isDoOnInput={false}
            max={9007199254740991}
            min={1}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyUp={[Function]}
            step={1}
            type="text"
            unit=""
          />
        </div>
      </div>
    </label>
  </label>
</div>
`;

exports[`should render correctly 2`] = `
<div
  className="tool-panel"
>
  <label
    className="controls accordion"
  >
    <input
      className="accordion-switcher"
      defaultChecked={true}
      type="checkbox"
    />
    <p
      className="caption"
    >
      Possible Rotation
      <span
        className="value"
      >
        2
      </span>
    </p>
    <label
      className="accordion-body"
    >
      <div>
        <div
          className="control nest-rotations"
        >
          <UnitInput
            abbr={false}
            className={Object {}}
            decimal={0}
            defaultValue={2}
            disabled={false}
            displayMultiValue={false}
            forceUsePropsUnit={false}
            getValue={[Function]}
            isDoOnInput={false}
            max={9007199254740991}
            min={1}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyUp={[Function]}
            step={1}
            type="text"
            unit=""
          />
        </div>
      </div>
    </label>
  </label>
</div>
`;
